packages/mattermost-redux/src/reducers/entities/threads.ts

```diff
@@ -57,6 +59,27 @@ export const threadsReducer = (state: ThreadsState['threads'] = {}, action: Gene
             },
         };
     }
+    case PostTypes.RECEIVED_NEW_POST: {
+        const post: Post = action.data;
+        const thread: UserThread | undefined = state[post.root_id];
+        if (post.root_id && thread) {
```

Comment:
```suggestion
        if (post.root_id && thread && !thread.is_following) {
```
If we check for not following, I think it would cut down on the number of double-updates as Followed threads additionally receive `SocketEvents.THREAD_UPDATED` which includes updated participants and reply_count.